server/app/app.py

```diff
@@ -143,6 +147,7 @@ def wrapped_function(self, dataset=None):
                 data_adaptor.set_uri_path(f"{self.url_dataroot}/{dataset}")
                 return func(self, data_adaptor)
         except (DatasetAccessError, DatasetNotFoundError, DatasetMetadataError) as e:
+            expire_metadata_cache(self.url_dataroot, dataset)
```

Comment:
I see that cache items will only be evicted if write lock can be acquired. Not sure if that's a practical concern, but might the cache need to delay eviction after write lock has been released? (e.g. marking the entry as `evict_requested` and using that upon lock release)

Reply:
If we get here, it means the dataset could not be loaded, and thus the other cache, `matrix_data_cache_manager`, should also not have this data cached. Maybe add a comment to that effect, unless you think it's brutally obvious. :)

Reply:
Out-of-scope for this PR, but might a tombstoned  dataset be removed from the `matrix_data_cache_manager`? Any need for maintaining its data? I assume it's worth keeping the _metadata_ cached for tombstoned datasets.

Reply:
re tombstoned datasets we want to store that information for redirects back to the dataportal in the case that someone finds the url for a deleted dataset from an external site (from a publication or other none data portal source)

Reply:
re eviction with lock, the metadata cache only stores the path to the dataset so it is unlikely that it will be locked. If by some chance it is, this code path is used for every endpoint so it will basically be called immediately after. But let me know if you think this is an antipattern or has the potential to cause issues

Reply:
I dontt think the call should raise issues but I can wrap it in a try except and log any errors just in case

Reply:
thx for explanations, no further concerns; adding the extra try/except sounds good.